### PR TITLE
Migrering skal kaste feil hvis simuleringen inneholder debittrekk, då…

### PR DESCRIPTION
**Pull request by blommish**
… det genererer en uventet utbetaling, som ikke skjer hvis vi venter me då migrere den til neste måned

https://favro.com/organization/98c34fb974ce445eac854de0/a64c6aad9b0d61ef6c0290bd?card=Tea-8344

### CODE REVIEW

File: src/main/kotlin/no/nav/familie/ef/sak/behandling/migrering/MigreringException.kt [hunk 1]
```diff
@@ -17,5 +17,6 @@ enum class MigreringExceptionType(val kanGåVidereTilJournalføring: Boolean = f
     ELDRE_PERIODER(kanGåVidereTilJournalføring = true),
     SIMULERING_FEILUTBETALING,
     SIMULERING_ETTERBETALING,
+    SIMULERING_DEBIT_TREKK,
```
Comment (bent-lorentzen) [inline src/main/kotlin/no/nav/familie/ef/sak/behandling/migrering/MigreringException.kt:20]: SIMULERING_DEBET_TREKK

File: src/main/kotlin/no/nav/familie/ef/sak/behandling/migrering/MigreringService.kt [hunk 2]
```diff
@@ -177,16 +180,29 @@ class MigreringService(
...
                                      MigreringExceptionType.SIMULERING_ETTERBETALING)
+        } else if (inneholderDebitTrekk(simulering)) {
+            throw MigreringException("Simuleringen inneholder posteringstypen TREKK med betalingstypen DEBIT. " +
+                                     "Dette blir en uønsket utbetaling pga en feil. Denne kan migreres på nytt i neste måned.",
+                                     MigreringExceptionType.SIMULERING_DEBIT_TREKK)
         }
     }
 
+    // Kan slettes når TØB fikset TOB-1739
+    private fun inneholderDebitTrekk(simulering: BeriketSimuleringsresultat) =
```
Comment (bent-lorentzen) [inline src/main/kotlin/no/nav/familie/ef/sak/behandling/migrering/MigreringService.kt:199]: inneholderDebettrekk
e og liten t

File: src/main/kotlin/no/nav/familie/ef/sak/behandling/migrering/MigreringService.kt [hunk 3]
```diff
@@ -177,16 +180,29 @@ class MigreringService(
...
+        if (oppsummering.feilutbetaling.compareTo(BigDecimal.ZERO) != 0) {
+            throw MigreringException("Feilutbetaling er ${oppsummering.feilutbetaling}",
                                      MigreringExceptionType.SIMULERING_FEILUTBETALING)
-        } else if (simulering.etterbetaling.compareTo(BigDecimal.ZERO) != 0) {
-            throw MigreringException("Etterbetaling er ${simulering.etterbetaling}",
+        } else if (oppsummering.etterbetaling.compareTo(BigDecimal.ZERO) != 0) {
+            throw MigreringException("Etterbetaling er ${oppsummering.etterbetaling}",
                                      MigreringExceptionType.SIMULERING_ETTERBETALING)
+        } else if (inneholderDebitTrekk(simulering)) {
+            throw MigreringException("Simuleringen inneholder posteringstypen TREKK med betalingstypen DEBIT. " +
```
Comment (bent-lorentzen) [inline src/main/kotlin/no/nav/familie/ef/sak/behandling/migrering/MigreringService.kt:192]: DEBET

Comment (blommish) [reply src/main/kotlin/no/nav/familie/ef/sak/behandling/migrering/MigreringService.kt]: [98ed5f4](https://github.com/navikt/familie-ef-sak/pull/1311/commits/98ed5f4df6900e7640b08e916c3a0388e3a2ce11)

File: src/main/kotlin/no/nav/familie/ef/sak/behandling/migrering/MigreringService.kt (hunk 2 above)
Comment (blommish) [reply src/main/kotlin/no/nav/familie/ef/sak/behandling/migrering/MigreringService.kt]: [98ed5f4](https://github.com/navikt/familie-ef-sak/pull/1311/commits/98ed5f4df6900e7640b08e916c3a0388e3a2ce11)

File: src/main/kotlin/no/nav/familie/ef/sak/behandling/migrering/MigreringException.kt (hunk 1 above)
Comment (blommish) [reply src/main/kotlin/no/nav/familie/ef/sak/behandling/migrering/MigreringException.kt]: [98ed5f4](https://github.com/navikt/familie-ef-sak/pull/1311/commits/98ed5f4df6900e7640b08e916c3a0388e3a2ce11)

**Review by charliemidtlyng — APPROVED** [review]

**Pull request merged**